ui-app/client/src/components/Portal.tsx

```diff
@@ -1,10 +1,11 @@
+import { isNullValue } from '@fincity/kirun-js';
 import React from 'react';
 import ReactDOM from 'react-dom';
-
-type Props = {
+interface Props {
 	children: React.ReactNode;
 	parent?: HTMLElement;
-};
+}
+
 const Portal: React.FC<Props> = ({ children, parent }) => {
 	const el = React.useMemo(() => document.createElement('div'), []);
```

Comment:
Don't use the div using the memo function. Create an element only if the parent is null. Add the containing div with the id. Use the memo function to generate a UUID to associate with the newly created div.

Reply:
On click of this div also the popover should be closed.